### PR TITLE
League stats module

### PR DESCRIPTION
**Pull request by cpfergus1**
Implemented league stats methods, all passing 100% coverage

### CODE REVIEW

File: lib/league_stats.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
+module LeagueStats
+  def count_of_teams
+    team_ids = @game_teams.map(&:team_id)
```
Comment (GregJMitchell) [inline lib/league_stats.rb:3]: Very short method. Great job

**Review by GregJMitchell — APPROVED** [review]
Looks Great. We can for sure shorten some of these methods on league_stats, but it looks great!

File: lib/game_teams.rb [hunk 2]
```diff
@@ -27,5 +28,6 @@ def initialize(row)
     @shots = @shots.to_i
     @tackles = @tackles.to_i
     @takeaways = @takeaways.to_i
+    @team_id = @team_id.to_i
```
Comment (s-steel) [inline lib/game_teams.rb:31]: We might run into issues with the data type for team_id and maybe other places.  I used team_id as a string.  We can fix that easily though.

File: lib/league_stats.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+       score_array.each do |pair|
+        if pair[0]==team
+          sum += pair[1]
+        end
+      end
+      avghash[team] = sum.to_f/(home_team_ids.count(team))
+    end
+    hash2 = Hash[teamnum.zip team_names]
+    require "pry"; binding.pry
+    hash2[avghash.key(avghash.values.min).to_i]
```
Comment (s-steel) [inline lib/league_stats.rb:135]: I like how the above methods are pretty easy to read up and down, we'll just have to refactor to get them smaller.

File: test/league_stats_test.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    assert_equal "Reign FC", @stat_tracker.highest_scoring_home_team
+  end
+
+  def test_lowest_scoring_visitor
+    assert_equal "Philadelphia Union", @stat_tracker.lowest_scoring_visitor
+  end
+
+  def test_lowest_scoring_home_team
+    assert_equal "Vancouver Whitecaps FC", @stat_tracker.lowest_scoring_home_team
+  end
```
Comment (s-steel) [inline test/league_stats_test.rb:45]: Tests look good.  I was thinking we should add multiple tests for each method.

**Review by s-steel — APPROVED** [review]
Nice work!  We will run into an issue with the data type for team_id because I used it as a different data type, but this will be an easy fix.  We'll also have to look at all the instance variables across all our branches to look for this issue.

**Pull request merged**